packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -368,6 +368,10 @@ function hydrateCustomElement(
     vnode: VCustomElement,
     renderer: RendererAPI
 ): Node | null {
+    if (elm.hasAttribute('data-lwc-skip-hydrate')) {
```

Comment:
```suggestion
    if (renderer.hasAttribute(elm, 'data-lwc-skip-hydrate')) {
```

This is our convention for abstracting differences between `engine-dom` and `engine-server`.